core/src/Streamly/Data/MutArray.hs

```diff
@@ -69,6 +69,12 @@ module Streamly.Data.MutArray
     -- * Size
     , length
 
+    -- * Unsafe functions
+    , getIndexUnsafe
+    , putIndexUnsafe
+    , modifyIndexUnsafe
```

Comment:
We should not group unsafe functions separately, it will be harder to find these. Should be put next to the similar safe functions.

Also can you check if the documentation of these is accurate.